src/frontend/src/flows/authenticate/index.ts

```diff
@@ -0,0 +1,355 @@
+import { initRegistration } from "../../components/registrationLink";
+import waitForAuthRequest, { AuthContext } from "./postMessageInterface";
+import { toggleErrorMessage } from "../../utils/errorHelper";
+import { fetchDelegation } from "./fetchDelegation";
+
+const pageContent = (
+  hostName: string,
+  editAnchor: boolean,
+  userNumber?: bigint
+) => html` <style>
+    .anchorText {
+      font-size: 1.5rem;
+    }
+
+    #userNumberInput {
+      margin: 0;
+      text-align: center;
+      width: 100%;
+      box-sizing: border-box;
+      padding: 0.5rem 2rem;
+    }
+
+    #userNumberInput:focus {
+      box-sizing: border-box;
+      border-style: double;
+      border-width: 2px;
+      border-radius: 4px;
+      border-image-slice: 1;
+      outline: none;
+      border-image-source: linear-gradient(
+        270.05deg,
+        #29abe2 10.78%,
+        #522785 22.2%,
+        #ed1e79 42.46%,
+        #f15a24 59.41%,
+        #fbb03b 77.09%
+      );
+    }
+
+    .modeContainer {
+      min-height: 6rem;
+    }
+
+    .childContainer {
+      position: relative;
+      margin-bottom: 0.5rem;
+    }
+
+    #editAnchorButton {
+      display: flex;
+      flex-direction: column;
+      position: absolute;
+      top: 0;
+      bottom: 0;
+      right: 0;
+      padding: 0.3rem;
+      margin: 0;
+      width: 2rem;
+      background: transparent;
+      border: none;
+    }
+
+    #registerButton {
+      margin: 2rem 0;
+    }
+
+    .hostName {
+      font-size: 1rem;
+      font-weight: 400;
+    }
+
+    .sectionTitle {
+      margin: 1rem 0 0.5rem 0;
+      color: black;
+    }
+
+    .buttonContainer {
+      margin: 0.5rem 0;
+    }
+
+    hr {
+      border-color: var(--grey-050);
+      margin: 1rem 0;
+    }
+  </style>
+  <div class="container">
+    <h1>Authorize Authentication</h1>
+    <p class="sectionTitle">Application URL</p>
+    <div class="highlightBox hostName">${hostName}</div>
+    <p class="sectionTitle">Identity Anchor</p>
+    <div>
+      ${!editAnchor && userNumber !== undefined
+        ? existingAnchorSection(userNumber)
+        : editAnchorSection(userNumber)}
+    </div>
+    <button type="button" id="authorizeButton" class="primary">
+      Authorize
+    </button>
+    <div id="registerSection" class="${editAnchor ? "" : "hidden"}">
+      <button type="button" id="registerButton">
+        Create New Identity Anchor
+      </button>
+    </div>
+    <div class="textLink">
+      <hr />
+      <div class="buttonContainer">
+        <button id="recoverButton" class="linkStyle">Lost access?</button>
+      </div>
+      <div class="buttonContainer">
+        <button type="button" class="linkStyle" id="manageButton">
+          Manage your Identity Anchor
+        </button>
+      </div>
+      <hr />
+    </div>
+    ${navbar}
+  </div>
+  ${footer}`;
+
+const existingAnchorSection = (userNumber: bigint) => html` <div
+  class="modeContainer"
+>
+  <div class="childContainer">
```

Comment:
nit: the two top-level divs (modeContainer, childContainer) could be lifted out of here. Also is there a way we could reuse the `input` (and maybe `disable` it) and just hide the `invalidAnchorMessage`? all that in order to modify the DOM as little as possible

Reply:
I lifted the `modeContainer` at least.